HHS/Head-Start-TTADP
# [TTAHUB-581] Update certain topics on legacy reports

## Description of change

Expand existing script to account for topics that either were missed or absent from the previous jira. Move it to the "tools" folder and add a CLI script to run it

Note - I chose to do this the *dumbest* possible way. It would have maybe been possible, maybe even advisable, to first strip all the pipe + abbreviations out of topics first (like "| ECS"), but that also would have required an additional step and to have had to account for all the variations in those as well, and I personally felt that since I already had the code and the list of things to change, it was easy enough to just directly stick all the different variations in the dictionary for updating

## How to test

If you have local data, you can run the following sql to get a count of all the offending reports in your DB (mine, a semi-recent export for staging had ~2000)
```sql
SELECT "id", "topics" FROM "ActivityReports" AS "ActivityReport" 
WHERE "ActivityReport"."submissionStatus" != 'deleted' 
AND "ActivityReport"."topics" && 
ARRAY[
'Behavioral / Mental Health',
'QIP',
'CLASS: Classroom Management',
'Curriculum (Early Childhood or Parenting)',
'Environmental Health and Safety',
'Behavioral / Mental Health | HS',
'Behavioral / Mental Health | HS, FES',
'Child Assessment, Development, Screening | ECS',
'Curriculum/a  (Instructional or Parenting) | ECS, FES',
'Curriculum (Early Chidhood or Parenting)',
'Eligibility (ERSEA) | ECS, FES, GS',
'Environmental Health and Safety | HS',
'Facilities | GS','Family Support Services | FES',
'Fiscal / Budget | GS','Human Resources | GS',
'Leadership / Governance | GS, FES',
'Nutrition | HS','Oral Health | HS',
'Parent and Family Engagement | ECS, FES, HS',
'Partnerships and Community Engagement | ECS, FES, SS',
'Physical Health and Screenings | HS',
'Pregnancy | FES, HS',
'Program Planning, Operations, Management Systems | GS',
'Quality Improvement / QIP | GS',
'Safety Practices | HS',
'Transition Practices | ECS',
'Transition Practices | ECS, FES',
'Transportation | GS',
'parent/family engagement/program planning/QIP'
]::VARCHAR(255)[];
```
Then running the ```yarn updateTopicNames:local``` script should yield console output as to what changed, and you can re-run the SQL to verify you have nothing remaining. 

It would probably be helpful to double check my dictionary of changes against the original Jira. I've double checked it now but another set of eyes probably wouldn't hurt.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-581

## Checklists

### Every PR



### Production Deploy


### After merge/deploy
